boot/boottest/device.go

```diff
@@ -46,30 +47,52 @@ func MockDevice(s string) snap.Device {
 		panic("MockDevice with no snap name and @mode is unsupported")
 	}
 	return &mockDevice{
-		bootSnap: bootsnap,
-		mode:     mode,
-		uc20:     uc20,
+		bootSnap:  bootsnap,
+		mode:      mode,
+		hasModes:  uc20,
+		isClassic: bootsnap == "",
 	}
 }
 
-// MockUC20Device implements boot.Device and returns true for HasModeenv.
-// Arguments are mode (empty means "run"), and model.
-// If model is nil a default model is used (same as MakeMockUC20Model).
-func MockUC20Device(mode string, model *asserts.Model) snap.Device {
+// MockDeviceWithModes implements boot.Device and returns true for
+// HasModeenv.  Arguments are mode (empty means "run"), model, and a
+// boolean specifying if this is a classic with modes or a UC device.
+// If model is nil a default model is used (MakeMockUC20Model or
+// MakeMockClassicWithModesModel is called).
+func MockDeviceWithModes(mode string, model *asserts.Model, isClassic bool) snap.Device {
```

Comment:
mmh, it seems this should be a not exported helper

Reply:
Right, I've unexported now